packages/x-charts/src/ChartsLocalizationProvider/ChartsLocalizationProvider.tsx

```diff
@@ -0,0 +1,88 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { useThemeProps } from '@mui/material/styles';
+import { ChartsLocaleText } from '../locales';
+import { CHARTS_DEFAULT_LOCALE_TEXT } from '../constants/defaultLocale';
+
+export interface ChartsLocalizationContextValue {
+  localeText: ChartsLocaleText;
+}
+
+export const ChartsLocalizationContext = React.createContext<ChartsLocalizationContextValue | null>(
+  null,
+);
+
+if (process.env.NODE_ENV !== 'production') {
+  ChartsLocalizationContext.displayName = 'ChartsLocalizationContext';
+}
+
+export interface ChartsLocalizationProviderProps {
+  children?: React.ReactNode;
+  /**
+   * Localized text for chart components.
+   */
+  localeText?: Partial<ChartsLocaleText>;
+}
+
+/**
+ * Demos:
+ *
+ * - [localization](https://mui.com/x/react-charts/localization/)
+ *
+ * API:
+ *
+ * - [ChartsLocalizationProvider API](https://mui.com/x/api/charts/charts-localization-provider/)
+ */
+function ChartsLocalizationProvider(inProps: ChartsLocalizationProviderProps) {
```

Comment:
What's the point of having a `ChartsLocalizationProvider`? I mean, why do we need this when we have the `ThemeProvider`?

I guess this is the unstyled version. But then, why don't we have a `LocalizationProvider` component that is shared between all components?